cEP-0022.md

```diff
@@ -0,0 +1,461 @@
+  will determine how
+  many inconsistencies were found in the project on assumming that particular
+  value of the setting.
+- On successfully finding a combination of setting values as *"green"*, a
+  `QUICKINFO.yaml` file will be generated storing the combination of setting
+  values along with the project Github links.
+
+### Example
+**`SpaceConsistencyBear`:**
+```python
+@deprecate_settings(indent_size='tab_width')
+def run(self,
+        filename,
+        file,
+        use_spaces: bool,
+        allow_trailing_whitespace: bool = False,
+        indent_size: int = SpacingHelper.DEFAULT_TAB_WIDTH,
+        enforce_newline_at_EOF: bool = True,
+        ):
+```
+Here `use_spaces`, `allow_trailing_whitespace`,
+`pydocstyle_add_ignore` and `enforce_newline_at_EOF` are Type 1 settings.
+
+
+## Type **2** or Type *infinite* settings
+
+### How will they be guessed?
+Since these settings can obtain any integer or other such value types, these
+settings will be
+guessed with the help of the **`QuickstartBear`**.
+
+### QuickstartBear
+This proposed bear does not generate any patches for the `file_dict` provided.
+It parses the `file_dict` and generates `QUICKDATA.yaml` of each and every
+particular value (except the ignored lines)
+each setting attains on every line of the project. Statistical
+data about the average, median and standard deviation of the values is also
+calculated and stored in `QUICKDATA.yaml` for further analysis.
+
+Most of these settings deal with the maximum limit of
+these values and a *"green"* value for the setting can be decided in an instant
+.
+
+### Metadata Generation
+- Suspected errors and a more suitable value of the setting will be guessed
+  by analysing the probability distribution functions of the setting values
+  and stored in `.quickfile` under the section `[amenable]`.
+
+### Example
+**`PEP8Bear`:**
+```python
+@deprecate_settings(indent_size='tab_width')
+def run(self, filename, file,
+        max_line_length: int = 79,
+        indent_size: int = SpacingHelper.DEFAULT_TAB_WIDTH,
+        pep_ignore: typed_list(str) = (),
+        pep_select: typed_list(str) = (),
+        local_pep8_config: bool = False,
+        ):
+```
+Here `max_line_length` is a Type 2 setting.
+
+## Type **3** or Type *discrete* settings
+
+### How will they be guessed?
+These settings will be guessed by using the metadata from applying Type
+Annotations to the setting values. (possibly using the
+[contracts](http://andreacensi.github.io/contracts/) package)
+The **Brute Force** or
+the **lite-mode** will run coala over and over again until it gets a
+*"green"* value for the setting.
+
+### Metadata Generation
+- In case of inability to find a *"green"* value for a setting the `.quickfile`
+  will store the setting along with a weight in section `[severe]`. The weight
+  will determine how
+  many inconsistencies were found in the project on assumming that particular
+  value of the setting.
```

Comment:
thats why for now I have kept the `SEVERITY` value thing, we never know how many errors should we display to the user. These values will be mapped to weights in the QuickFile after some analysis is done on the project. Also since I was asked to remove the threshold feature, I would be a hard thing to do. we really could use some values from the project url to remap the `SEVERITY` value to the weights but based on the reputation of the org. Pretty doubtful over here. I could add a TODO kind of thing here to be appended later on until something is finalised